base/show.jl

```diff
@@ -322,6 +327,11 @@ function isidentifier(s::AbstractString)
     end
     return true
 end
+"""
+    isidentifier(s) -> Bool
+
+Returns whether symbol is vaild identifier.
```

Comment:
We should probably explain the difference between a symbol and an identifier here. I find it hard to explain myself, but in order to try to be helpfull, I'll make an attempt.

```md
Returns whether symbol contianis charaters that is parsed as a valid identifier in Julia code.

Internally Julia allows any sequence of characters in a `Symbol` (exept`\0`s),
and macros automatically uses variable names containing `#` in order to
avoid naming collision with the surrounding code. In order for the parser to
recognize a variable, it uses a limited set of characters (greatly extended by Unicode).
``isidentifier()`` makes it possible to query the parser directly whether a symbol contains
valid characters.
```